### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by rmoreliovlabs**
## Description
This time we are trying to add e2e-like tests where we can start the node using regtest db and ensure the node behavior. We are going to start testing CLI tools as the first try.

* Added a new Gradle Task for `integrationTest`
* Added an `integrationTest` folder with necessary resources and and java tests.
* Created a new integration test step for rskj-tests job in CircleCI. 

## Motivation and Context
We don't have a proper way to check if tool actually works. We need to a way to simulate the complete usage of the tool in order to confirm the correct behavior of the tool.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
           command: |
             ./configure.sh
             ./gradlew --no-daemon dependencies
-            ./gradlew --no-daemon --stacktrace build -x test
+            ./gradlew --no-daemon --stacktrace build -x test -x integrationTest
```
Comment (Vovchyk) [inline .circleci/config.yml:37]: shouldn't we exclude int. tests in this [line](https://github.com/rsksmart/rskj/blob/f35bf63c34fac76acbbe59cd27773572ae5df204/.circleci/config.yml#L63) too?

File: rskj-core/src/integrationTest/resources/redeemScripts.json
Comment (Vovchyk) [inline rskj-core/src/integrationTest/resources/redeemScripts.json:1]: it looks like this json file is only used in the unit tests. Do we need it?

File: rskj-core/src/integrationTest/resources/redeemScripts_p2shERP.json
Comment (Vovchyk) [inline rskj-core/src/integrationTest/resources/redeemScripts_p2shERP.json:1]: it looks like this json file is only used in the unit tests. Do we need it?

File: rskj-core/src/main/java/co/rsk/cli/tools/StartBootstrap.java [hunk 2]
```diff
@@ -74,6 +77,16 @@ public Integer call() throws IOException {
         NodeStopper nodeStopper = System::exit;
 
         runBootstrapNode(ctx, preflightChecks, runtime, nodeStopper);
+
+        if (this.stopTimeout > 0L) {
+            try {
+                Thread.sleep(this.stopTimeout);
```
Comment (Vovchyk) [inline rskj-core/src/main/java/co/rsk/cli/tools/StartBootstrap.java:83]: could you please explain why we need to "sleep()" here?

**Review by Vovchyk — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (rmoreliovlabs) [reply .circleci/config.yml:37]: Done

File: rskj-core/src/integrationTest/resources/redeemScripts.json
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/resources/redeemScripts.json:1]: Removed them

File: rskj-core/src/integrationTest/resources/log4j.properties
[the remark this answers is not in the document]
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/resources/log4j.properties:1]: Removed

File: rskj-core/src/integrationTest/resources/redeemScripts_p2shERP.json
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/resources/redeemScripts_p2shERP.json:1]: Removed

File: rskj-core/src/integrationTest/resources/rskj.conf
[the remark this answers is not in the document]
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/resources/rskj.conf:1]: No it is not. Already deleted

File: rskj-core/src/main/java/co/rsk/cli/tools/StartBootstrap.java (hunk 2 above)
Comment (rmoreliovlabs) [reply rskj-core/src/main/java/co/rsk/cli/tools/StartBootstrap.java:83]: Just for testing purposes. Right now, when we run the Bootstrap tool, the main thread ends fast and then close the rsk context, making the tool fail when starting the UPD server. So I can't make sure the Bootstrap node really started. So I just added a small timeout to make it wait some time so I can make necessary checks before the node is shut down.

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Comment by rmoreliovlabs** [comment]
pipeline:run

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,599 @@
...
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class CliToolsIntegrationTest {
+    private String projectPath;
+    private String buildLibsPath;
+    private String jarName;
+    private String databaseDir;
+    private String bloomsDbDir;
+    private final JsonNodeFactory jsonNodeFactory = JsonNodeFactory.instance;
```
Comment (Vovchyk) [inline rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:61]: not used?

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,599 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2017 RSK Labs Ltd.
```
Comment (Vovchyk) [inline rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:3]:
```suggestion
 * Copyright (C) 2023 RSK Labs Ltd.
```

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,599 @@
...
+    private String projectPath;
+    private String buildLibsPath;
+    private String jarName;
+    private String databaseDir;
+    private String bloomsDbDir;
+    private final JsonNodeFactory jsonNodeFactory = JsonNodeFactory.instance;
+    private final int port = 9999;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private String[] baseArgs;
+    private LinkedList<String> lnkListBaseArgs;
```
Comment (Vovchyk) [inline rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:65]: not used?

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,599 @@
...
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class CliToolsIntegrationTest {
+    private String projectPath;
```
Comment (Vovchyk) [inline rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:56]: Field can be converted to a local variable

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,599 @@
...
+    private String bloomsDbDir;
+    private final JsonNodeFactory jsonNodeFactory = JsonNodeFactory.instance;
+    private final int port = 9999;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private String[] baseArgs;
+    private LinkedList<String> lnkListBaseArgs;
+    private String strBaseArgs;
+    private String integrationTestResourcesPath;
+    private String logbackXmlFile;
+    private String rskConfFile;
```
Comment (Vovchyk) [inline rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:69]: Fields can be converted to a local variable

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 8]
```diff
@@ -0,0 +1,599 @@
...
+    private String baseJavaCmd;
+
+
+    class CustomProcess {
```
Comment (Vovchyk) [inline rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:73]: static ?

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 9]
```diff
@@ -0,0 +1,599 @@
...
+        private final Process process;
```
Comment (Vovchyk) [inline rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:74]: not used?

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 10]
```diff
@@ -0,0 +1,599 @@
...
+        private final String input;
+        private final String errors;
```
Comment (Vovchyk) [inline rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:76]: not used?

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 11]
```diff
@@ -0,0 +1,599 @@
...
+        public String getErrors() {
+            return errors;
+        }
+    }
+
+    @TempDir
+    private Path tempDir;
+
+    private String getProcStreamAsString(InputStream in) throws IOException {
+        byte bytesAvailable[] = new byte[in.available()];
```
Comment (Vovchyk) [inline rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:101]:
```suggestion
        byte[] bytesAvailable = new byte[in.available()];
```

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 12]
```diff
@@ -0,0 +1,599 @@
...
+        if (beforeDestroyFn != null) {
+            beforeDestroyFn.accept(proc);
+        }
+
+        proc.destroy();
+
+        return new CustomProcess(proc, procInput, procErrors);
+    }
+
+    private String getHashFromLog(List<String> logLines, int logIndex) {
```
Comment (Vovchyk) [inline rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:153]: not used?

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 13]
```diff
@@ -0,0 +1,599 @@
...
+        runCommand(cmd, 1, TimeUnit.MINUTES);
+
+        FileUtil.recursiveDelete(databaseDir);
+
+        cmd = String.format("%s -cp %s/%s co.rsk.cli.tools.ImportBlocks --file %s %s", baseJavaCmd, buildLibsPath, jarName, blocksFile.getAbsolutePath(), strBaseArgs);
+        runCommand(cmd, 1, TimeUnit.MINUTES);
+
+        RskContext rskContext = new RskContext(baseArgs);
+
+        Long maxNumber = rskContext.getBlockStore().getMaxNumber();
```
Comment (Vovchyk) [inline rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:437]:
```suggestion
        long maxNumber = rskContext.getBlockStore().getMaxNumber();
```

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 14]
```diff
@@ -0,0 +1,599 @@
...
+            when(block.getHash()).thenReturn(blockHash);
+            when(block.getParentHash()).thenReturn(parentHash);
+            when(block.getNumber()).thenReturn(i);
+            when(block.getEncoded()).thenReturn(bestBlock.getEncoded());
+
+            rskContext.getBlockStore().saveBlock(block, BlockDifficulty.ZERO, true);
+            parentHash = blockHash;
+        }
+
+        Long maxNumber = rskContext.getBlockStore().getMaxNumber();
```
Comment (Vovchyk) [inline rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:522]:
```suggestion
        long maxNumber = rskContext.getBlockStore().getMaxNumber();
```

File: rskj-core/src/integrationTest/resources/integration-test-rskj.conf [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+        # if peer discovery is off
+        # the peer window will show
+        # only what retrieved by active
+        # peer [true/false]
+        enabled = true
+
+        # List of the peers to start
+        # the search of the online peers
+        # values: [ip:port]
+        ip.list = [
```
Comment (Vovchyk) [inline rskj-core/src/integrationTest/resources/integration-test-rskj.conf:14]: is there a need to use mainnet bootstraps in the int. tests? Can this list, for instance, be empty?

**Review by Vovchyk — CHANGES_REQUESTED** [review]

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java (hunk 4 above)
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:3]: Addressed

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java (hunk 5 above)
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:65]: Addressed

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java (hunk 7 above)
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:69]: Addressed

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java (hunk 9 above)
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:74]: Addressed

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java (hunk 8 above)
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:73]: addressed

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java (hunk 6 above)
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:56]: addressed

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java (hunk 11 above)
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:101]: Addressed

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java (hunk 13 above)
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:437]: Addressed

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java (hunk 14 above)
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:522]: Addressed

File: rskj-core/src/integrationTest/resources/integration-test-rskj.conf (hunk 15 above)
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/resources/integration-test-rskj.conf:14]: Addressed

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 16]
```diff
@@ -0,0 +1,568 @@
...
+
+    private CustomProcess runCommand(String cmd, int timeout, TimeUnit timeUnit) throws InterruptedException, IOException {
+        return runCommand(cmd, timeout, timeUnit, null);
+    }
+
+    private CustomProcess runCommand(String cmd, int timeout, TimeUnit timeUnit, Consumer<Process> beforeDestroyFn) throws InterruptedException, IOException {
+        Process proc = Runtime.getRuntime().exec(cmd);
+
+        proc.waitFor(timeout, timeUnit);
+        String procInput = getProcStreamAsString(proc.getInputStream());
```
Comment (Vovchyk) [inline rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:123]:
```suggestion
        String procOutput = getProcStreamAsString(proc.getInputStream());
```

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 17]
```diff
@@ -0,0 +1,568 @@
...
+    private String bloomsDbDir;
+    private final int port = 9999;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private String[] baseArgs;
+    private String strBaseArgs;
+    private String baseJavaCmd;
+
+
+    static class CustomProcess {
+        private final String input;
```
Comment (Vovchyk) [inline rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:68]:
```suggestion
        private final String output;
```

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 18]
```diff
@@ -0,0 +1,568 @@
...
+
+        public CustomProcess(String input) {
+            this.input = input;
+        }
+
+        public String getInput() {
```
Comment (Vovchyk) [inline rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:74]:
```suggestion
        public String getOutput() {
```

**Review by Vovchyk — CHANGES_REQUESTED** [review]

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 19]
```diff
@@ -0,0 +1,568 @@
...
+
+import co.rsk.RskContext;
+import co.rsk.core.BlockDifficulty;
+import co.rsk.crypto.Keccak256;
+import co.rsk.trie.Trie;
+import co.rsk.util.HexUtils;
+import co.rsk.util.TestUtils;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
```
[the remark this answers is not in the document]
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:28]: Addressed

File: rskj-core/src/integrationTest/java/co/rsk/util/TestUtils.java [hunk 20]
```diff
@@ -0,0 +1,34 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2017 RSK Labs Ltd.
```
[the remark this answers is not in the document]
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/util/TestUtils.java:3]: Addressed

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 21]
```diff
@@ -0,0 +1,568 @@
...
+        public String getInput() {
+            return input;
+        }
+
+    }
+
+    @TempDir
+    private Path tempDir;
+
+    private String getProcStreamAsString(InputStream in) throws IOException {
```
[the remark this answers is not in the document]
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:83]: Addressed

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 22]
```diff
@@ -0,0 +1,568 @@
...
+                "        true\n" +
+                "    ],\n" +
+                "    \"id\": 1,\n" +
+                "    \"jsonrpc\": \"2.0\"\n" +
+                "}]";
+
+        return sendJsonRpcMessage(content);
+    }
+
+    private static OkHttpClient getUnsafeOkHttpClient() {
```
[the remark this answers is not in the document]
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:148]: Done

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java (hunk 21 above)
[the remark this answers is not in the document]
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:83]: Sounds good. Addressed

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 23]
```diff
@@ -0,0 +1,568 @@
...
+                        public boolean verify(String hostname, SSLSession session) {
+                            return true;
+                        }
+                    });
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Response sendJsonRpcMessage(String content) throws IOException {
```
[the remark this answers is not in the document]
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:187]: Sounds good! Done

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 24]
```diff
@@ -0,0 +1,568 @@
...
+        runCommand(
+                cmd,
+                1,
+                TimeUnit.MINUTES,
+                proc -> {
+                    try {
+                        Response response = getBestBlock();
+                        responseMap.put("latestProcessedBlock", response);
+                    } catch (IOException e) {
+                        e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:210]: Addressed

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 25]
```diff
@@ -0,0 +1,568 @@
...
+        };
+        strBaseArgs = String.join(" ", baseArgs);
+        baseJavaCmd = String.format("java %s %s", String.format("-Dlogback.configurationFile=%s", logbackXmlFile), String.format("-Drsk.conf.file=%s", rskConfFile));
+    }
+
+    private CustomProcess runCommand(String cmd, int timeout, TimeUnit timeUnit) throws InterruptedException, IOException {
+        return runCommand(cmd, timeout, timeUnit, null);
+    }
+
+    private CustomProcess runCommand(String cmd, int timeout, TimeUnit timeUnit, Consumer<Process> beforeDestroyFn) throws InterruptedException, IOException {
```
[the remark this answers is not in the document]
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:119]: Addressed

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java [hunk 26]
```diff
@@ -0,0 +1,568 @@
...
+        String cmd = String.format("%s -cp %s/%s co.rsk.Start --reset %s", baseJavaCmd, buildLibsPath, jarName, strBaseArgs);
+        runCommand(
+                cmd,
+                1,
+                TimeUnit.MINUTES, proc -> {
+                    try {
+                        Response response = getBestBlock();
+                        responseMap.put("latestProcessedBlock", response);
+                    } catch (IOException e) {
+                        e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:285]: Addressed

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java (hunk 16 above)
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:123]: Addressed

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java (hunk 17 above)
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:68]: Addressed

File: rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java (hunk 18 above)
Comment (rmoreliovlabs) [reply rskj-core/src/integrationTest/java/co/rsk/cli/tools/CliToolsIntegrationTest.java:74]: Addressed

File: rskj-core/src/integrationTest/resources/integration-test-rskj.conf [hunk 27]
```diff
@@ -0,0 +1,16 @@
+peer {
+
+    discovery = {
+
+        # if peer discovery is off
+        # the peer window will show
+        # only what retrieved by active
+        # peer [true/false]
+        enabled = true
```
Comment (Vovchyk) [inline rskj-core/src/integrationTest/resources/integration-test-rskj.conf:9]:
```suggestion
        enabled = false
```
one more thing spotted.. probably in the `regtest` mode we don't need discovery to be `enabled`

**Review by Vovchyk — CHANGES_REQUESTED** [review]

**Review by Vovchyk — APPROVED** [review]

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Pull request merged**